File: apprefurbished/src/main/java/com/androidcollider/easyfin/debts/list/DebtsPresenter.java
package com.androidcollider.easyfin.debts.list;

import android.support.annotation.Nullable;

import com.androidcollider.easyfin.common.models.Debt;

import java.util.List;

import rx.Subscriber;

/**
 * @author Ihor Bilous
 */

class DebtsPresenter implements DebtsMVP.Presenter {

    @Nullable
    private DebtsMVP.View view;
    private DebtsMVP.Model model;


    DebtsPresenter(DebtsMVP.Model model) {
        this.model = model;
    }

    @Override
    public void setView(@Nullable DebtsMVP.View view) {
        this.view = view;
    }

    @Override
    public void loadData() {
        model.getDebtList()
                .subscribe(new Subscriber<List<DebtViewModel>>() {

                    @Override
                    public void onCompleted() {

                    }

                    @Override
                    public void onError(Throwable e) {

                    }

                    @Override
                    public void onNext(List<DebtViewModel> debtList) {
                        if (view != null) {
                            view.setDebtList(debtList);
                        }
                    }
                });
    }

    @Override
    public void getDebtById(int id, int mode, int actionType) {
        model.getDebtById(id)
                .subscribe(new Subscriber<Debt>() {

                    @Override
                    public void onCompleted() {

                    }

                    @Override
                    public void onError(Throwable e) {

                    }

                    @Override
                    public void onNext(Debt debt) {
                        if (view != null) {
                            switch (actionType) {
                                case DebtsFragment.ACTION_EDIT:
                                    view.goToEditDebt(debt, mode);
                                    break;
                                case DebtsFragment.ACTION_PAY:
                                    view.goToPayDebt(debt, mode);
                                    break;
                            }
                        }
                    }
                });
    }

    @Override
    public void deleteDebtById(int id) {
        model.deleteDebtById(id)
                .subscribe(new Subscriber<Boolean>() {

                    @Override
                    public void onCompleted() {

                    }

                    @Override
                    public void onError(Throwable e) {

                    }

                    @Override
                    public void onNext(Boolean aBoolean) {
                        if (aBoolean && view != null) {
                            view.deleteDebt();
                        }
                    }
                });
    }
}

File: apprefurbished/src/main/java/com/androidcollider/easyfin/debts/list/DebtsPresenter.kt
package com.androidcollider.easyfin.debts.list

import com.androidcollider.easyfin.common.models.Debt

/**
 * @author Ihor Bilous
 */
internal class DebtsPresenter(private val model: DebtsMVP.Model) : DebtsMVP.Presenter {
    private var view: DebtsMVP.View? = null
    override fun setView(view: DebtsMVP.View?) {
        this.view = view
    }

    override fun loadData() {
        model.debtList
            .subscribe(
                { debtList: List<DebtViewModel> -> view?.setDebtList(debtList) },
                { obj: Throwable -> obj.printStackTrace() })
    }

    override fun getDebtById(id: Int, mode: Int, actionType: Int) {
        model.getDebtById(id)
            .subscribe(
                { debt: Debt ->
                    when (actionType) {
                        DebtsFragment.ACTION_EDIT -> view?.goToEditDebt(debt, mode)
                        DebtsFragment.ACTION_PAY -> view?.goToPayDebt(debt, mode)
                    }
                },
                { obj: Throwable -> obj.printStackTrace() })
    }

    override fun deleteDebtById(id: Int) {
        model.deleteDebtById(id)
            .subscribe(
                { aBoolean: Boolean ->
                    if (aBoolean) {
                        view?.deleteDebt()
                    }
                },
                { obj: Throwable -> obj.printStackTrace() })
    }
}